test/functionalTest/cases/722_unhardwire_format_cpr/notify_format_in_ngsi9_subscriptions.test

```diff
@@ -0,0 +1,944 @@
+# Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Subscriptions For NGSI9
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+accumulatorStart
+
+--SHELL--
+
+#
+# The format for Subscriptions are sent in either JSON or XML, depending on two things:
+# 1. The URI parameter notifyFormat
+# 2. The format that the subscription in sent in
+#
+# The URI parameter notifyFormat (1) overrides format that the subscription in sent in (2).
+#
+# So, we have a number of combinations here:
+# 1. Create a subscription in JSON, without using URI parameter notifyFormat - see notification in JSON
+# 2. Create a subscription in XML, without using URI parameter notifyFormat - see notification in XML
+# 3. Create a subscription in JSON, with URI parameter notifyFormat in JSON - see notification in JSON
+# 4. Create a subscription in JSON, with URI parameter notifyFormat in XML - see notification in XML
+# 5. Create a subscription in XML, with URI parameter notifyFormat in JSON - see notification in JSON
+# 6. Create a subscription in XML, with URI parameter notifyFormat in XML - see notification in XML
+#
+# 
+# TEST PLAN:
+# 0.1. Availability-Subscribe to E1/A0 in JSON, with invalid notifyFormat=ABCD
+# 0.2. Availability-Subscribe to E1/A0 in XML, with invalid notifyFormat=ABCD
+#
+# 1.1. Register E1/T1/A1
+# 1.2. Availability-Subscribe to E1/A1 in JSON 
+# 1.3. Dump accumulator to see notification of A1 in JSON, as the subscription was done in JSON
+# 1.4. Reset the accumulator
+#
+#   1.5. Modify subscription in XML (later)
```

Comment:
Yes. I wanted to add that also to make it 100% complete but it didn't work and I didn't really have the time ...